src/eventarc/eventarc-client-internal.ts

```diff
@@ -0,0 +1,156 @@
+/*!
+ * @license
+ * Copyright 2022 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as validator from '../utils/validator';
+import { FirebaseEventarcError, toCloudEventProtoFormat } from './eventarc-utils';
+import { Eventarc, Channel } from './eventarc';
+import {
+  HttpRequestConfig, HttpClient, HttpError, AuthorizedHttpClient
+} from '../utils/api-request';
+import { FirebaseApp } from '../app/firebase-app';
+import * as utils from '../utils';
+import { PrefixedFirebaseError } from '../utils/error';
+import { CloudEvent } from './cloudevent';
+
+const EVENTARC_API = 'https://eventarcpublishing.googleapis.com/v1';
+const FIREBASE_VERSION_HEADER = {
+  'X-Firebase-Client': `fire-admin-node/${utils.getSdkVersion()}`,
+};
+const CHANNEL_NAME_REGEX = /^(projects\/([^/]+)\/)?locations\/([^/]+)\/channels\/([^/]+)$/;
+const DEFAULT_CHANNEL_REGION = 'us-central1';
+
+/**
+ * Class that facilitates sending requests to the Eventarc backend API.
+ *
+ * @internal
+ */
+export class EventarcApiClient {
+  private readonly httpClient: HttpClient;
+  private projectId?: string;
+  private readonly resolvedChannelName: Promise<string>;
+
+  constructor(private readonly eventarc: Eventarc, private readonly channel: Channel) {
+    if (!validator.isNonNullObject(eventarc.app) || !('options' in eventarc.app)) {
+      throw new FirebaseEventarcError(
+        'invalid-argument',
+        'First argument passed to admin.appCheck() must be a valid Firebase app instance.');
```

Comment:
If we just pass the `App` instead of an `Eventarc` instance, this error message would make more sense.
Typo: admin.appCheck()

Reply:
Done.